cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/queue/hbase/HBaseQueueAdmin.java

```diff
@@ -567,4 +628,26 @@ private void createQueueTable(TableDescriptorBuilder tdBuilder, byte[][] splitKe
       }
     }
   }
+
+  @SuppressWarnings("ThrowableResultOfMethodCallIgnored")
+  private Map<TableId, Throwable> waitForUpgrade(Map<TableId, Future<?>> upgradeFutures) throws InterruptedException {
+    Map<TableId, Throwable> failed = new HashMap<>();
+    for (Map.Entry<TableId, Future<?>> entry : upgradeFutures.entrySet()) {
+      while (!entry.getValue().isDone()) {
```

Comment:
I was planning on adding an overall timeout, but decided not to do it now. I'll remove the loop